### PR TITLE
Add support code and tests for TLSIdentity

### PR DESCRIPTION
**Pull request by pasin**
* Added TLSIdentityTest
* Added Listener logging domain
* As the Keychain cannot be accessed by tests if the tests are not running under a host app, added hasHostApp property to the CBLTestCase to check whether the tests are currently running under a host app or not.

#CBL-816

### CODE REVIEW

File: Objective-C/Tests/TLSIdentityTest.m [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    CFErrorRef errRef = nil;
+    NSData* publicKeyData = CFBridgingRelease(SecKeyCopyExternalRepresentation(publicKeyRef, &errRef));
+    Assert(errRef == nil);
+    AssertNotNil(publicKeyData);
+    Assert(publicKeyData.length > 0);
+    NSLog(@"%lu", (unsigned long)publicKeyData.length);
+}
+
+- (void) deleteAllItemsInKeyChain {
+    NSArray *classes = @[(id)kSecClassKey, (id)kSecClassCertificate, (id)kSecClassIdentity];
```
Comment (jayahariv) [inline Objective-C/Tests/TLSIdentityTest.m:106]: is this going to delete all certificates and identities form keychain?

File: Objective-C/Tests/TLSIdentityTest.m [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    
+    // Create again with the same label:
+    identity = [CBLTLSIdentity createIdentityForServer: YES
+                                            attributes: attrs
+                                            expiration: nil
+                                                 label: kServerCertLabel
+                                                 error: &error];
+    
+    AssertNil(identity);
+    AssertNotNil(error);
```
Comment (jayahariv) [inline Objective-C/Tests/TLSIdentityTest.m:220]: AssertEqual(error.code, kC4ErrorCrypto);
error.localisedDescription.contains("OSStatus = -25299") ?

File: Objective-C/Tests/TLSIdentityTest.m [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    
+    // Create again with the same label:
+    identity = [CBLTLSIdentity createIdentityForServer: YES
+                                            attributes: attrs
+                                            expiration: nil
+                                                 label: kClientCertLabel
+                                                 error: &error];
+    
+    AssertNil(identity);
+    AssertNotNil(error);
```
Comment (jayahariv) [inline Objective-C/Tests/TLSIdentityTest.m:296]: check error code and osstatus code?

File: Objective-C/Tests/TLSIdentityTest.m (hunk 1 above)
Comment (pasin) [reply Objective-C/Tests/TLSIdentityTest.m:106]: It's fine for iOS as the keychain items are app's bound. For macOS, there could be some side effects. Will try to improve this.

File: Objective-C/Tests/TLSIdentityTest.m [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+}
+
+- (void) testCreateIdentityFromData {
+    if (!self.hasHostApp) return;
+    
+    NSData* data = [self dataFromResource: @"identity/certs" ofType: @"p12"];
+    
+    // When importing P12 file on macOS unit test, there is an internal exception thrown
+    // inside SecPKCS12Import() which doesn't actually cause anything. Ignore the exception
+    // so that the exception brekpoint will not be triggered.
```
Comment (jayahariv) [inline Objective-C/Tests/TLSIdentityTest.m:306]: typo.  breakpoint

File: Objective-C/Tests/TLSIdentityTest.m (hunk 1 above)
Comment (pasin) [reply Objective-C/Tests/TLSIdentityTest.m:106]: Actually that method is not used anymore. Forgot that I already changed to delete the identity directly which is definitely safer.

File: Objective-C/Tests/TLSIdentityTest.m (hunk 2 above)
Comment (pasin) [reply Objective-C/Tests/TLSIdentityTest.m]: Done

**Pull request merged**